### PR TITLE
Operating computers no longer link to stasis beds.

### PR DESCRIPTION
**Pull request by KathyRyals**
## About The Pull Request

Operating computers no longer link to stasis beds.

## Why It's Good For The Game

Medical literally always moves the computers to the stasis beds, bypassing operating tables almost completely. Stasis beds are not meant to be THE medical tool, yet they currently are. With this change, operating tables and stasis beds will each have their own niche, and it will hopefully provide a bit more diversity in how medical treat patients.

With approval from @ExcessiveUseOfCobblestone 

## Changelog
:cl:
del: Operating computers no longer link to stasis beds. 
/:cl:

### CODE REVIEW

**Comment by Sylphet** [comment]
_grabs popcorn_

**Comment by Time-Green** [comment]
On one hand, you're right. It forces medical doctors to rush operating consoles because they'll be at a horrible disadvantage otherwise.

On the other hand, this forces medical doctors to make boatloads of formaldehyde to ever have a hope of treating badly damaged sick patients. And even in your average medbay round, the surgery rooms are still used plenty for the occational non-emergency surgery. This change will just make medical doctors have to constantly move the bodies if they ever want a chance to revive someone, which could be good if medbay gets overcrowded. It'd also be great so that I can revive someone without the horde of three assistants that are always in medbay stunning me and looting the HoPs body

In short: I know it's better in the long term but fuck man. Just merge it and get it over with

**Comment by hermitec** [comment]
Sleepers 2 stasis boogaloo

**Comment by Mickyan** [comment]
As with every time a slapdash stasis nerf comes up this elevates letting your patients die and filling them with formaldehyde as the preferred method to triage and treatment since it's mechanically equivalent to having stasis anywhere (also the patients don't complain as much when they're dead)

I'm pretty sure at some point Cobby mentioned the wish to detach advanced surgeries from surgery computers so I'm not sure what angle they're playing here

**Comment by D4C-420** [comment]
all I see this doing is slowing down tend wounds which is the largest reason for moving the surgery computers to stasis beds. The advantage of having time to fix someone compared to them dying of blood loss or toxin, or oxy is much larger than doing surgery 10% faster.

**Comment by Fikou** [comment]
> all I see this doing is slowing down tend wounds which is the largest reason for moving the surgery computers to stasis beds. The advantage of having time to fix someone compared to them dying of blood loss or toxin, or oxy is much larger than doing surgery 10% faster.

but its not 10% faster though, it gets much faster with every research and there will be a reason to use surgery instead of stasis, for those huge speed upgrades you get later in the round with research

**Comment by CoffeeDragon16** [comment]
> On the other hand, this forces medical doctors to make boatloads of formaldehyde to ever have a hope of treating badly damaged sick patients.
I don't find this to be true, as you can just jab someone with their backpack medipen to preserve them. It's much faster, and I see that used more often.

**Comment by D4C-420** [comment]
> but its not 10% faster though, it gets much faster with every research and there will be a reason to use surgery instead of stasis, for those huge speed upgrades you get later in the round with research

I was referring to the -10% speed for steps on stasis beds. Forgot it made it faster. Still point applies this would just slow down doing tend wounds more than anything.

**Comment by Thalpy** [comment]
What if to balance this formaldehyde didn't require silver in it's creation? Actually I've got some old code that I can use as a medium between the two that would work better. Lemme put up a PR and link it.

**Comment by D4C-420** [comment]
> What if to balance this formaldehyde didn't require silver in it's creation? Actually I've got some old code that I can use as a medium between the two that would work better. Lemme put up a PR and link it.

that or as suggested by https://github.com/tgstation/tgstation/pull/54412 make silver a catalyst for formaldehyde which puts it in line with every other reaction which uses raw ore.

**Comment by antropod** [comment]
1. put stasis next to operating table
2. move patient to operating table, start surgery
3. move back to stasis
4. ...
5. profit

**Comment by jwc1015** [comment]
Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.

**Comment by antropod** [comment]
just borg him (c)

**Comment by Time-Green** [comment]
> Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.

You can still perform surgery on a stasis beds. All you need is revival surgery, which you can perform without operating console. You can also just give them a heart for some reason and that just works aswell

**Comment by antropod** [comment]
you can defib plasmamen without implanting heart

**Comment by jwc1015** [comment]
> > Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.
> 
> You can still perform surgery on a stasis beds. All you need is revival surgery, which you can perform without operating console. You can also just give them a heart for some reason and that just works aswell

There ARE various surgeries that need an operating computer and stasis bed; those are now almost impossible without a plasma room or mediborg

**Comment by antropod** [comment]
tend wounds can be performed without stripping
organ manipulation is basic surgery, can be performed on stasis bed
brain surgery is basic surgery

**Comment by Time-Green** [comment]
> > > Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.
> > 
> > 
> > You can still perform surgery on a stasis beds. All you need is revival surgery, which you can perform without operating console. You can also just give them a heart for some reason and that just works aswell
> 
> There ARE various surgeries that need an operating computer and stasis bed; those are now almost impossible without a plasma room or mediborg

I misread, you're right. You have a good point. What works the best for me is running to a surgery table with operating console, starting the surgery, and then rushing him back to stasis to finish the surgery. But you're right, this is kind of an issue depending on how much people want to fuck over plasmamen

**Comment by jwc1015** [comment]
> > > Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.
> > 
> > 
> > You can still perform surgery on a stasis beds. All you need is revival surgery, which you can perform without operating console. You can also just give them a heart for some reason and that just works aswell
> 
> There ARE various surgeries that need an operating computer and stasis bed; those are now almost impossible without a plasma room or mediborg



> > > > Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.
> > > 
> > > 
> > > You can still perform surgery on a stasis beds. All you need is revival surgery, which you can perform without operating console. You can also just give them a heart for some reason and that just works aswell
> > 
> > 
> > There ARE various surgeries that need an operating computer and stasis bed; those are now almost impossible without a plasma room or mediborg
> 
> I misread, you're right. You have a good point. What works the best for me is running to a surgery table with operating console, starting the surgery, and then rushing him back to stasis to finish the surgery. But you're right, this is kind of an issue depending on how much people want to fuck over plasmamen

Admittedly, said surgeries aren’t vital to MEDBAY, but they’re still nigh-impossible for plasmamen without quite a bit of setup beforehand.

**Comment by jwc1015** [comment]
> > > > > Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.
> > > > 
> > > > 
> > > > You can still perform surgery on a stasis beds. All you need is revival surgery, which you can perform without operating console. You can also just give them a heart for some reason and that just works aswell
> > > 
> > > 
> > > There ARE various surgeries that need an operating computer and stasis bed; those are now almost impossible without a plasma room or mediborg
> > 
> > 
> > I misread, you're right. You have a good point. What works the best for me is running to a surgery table with operating console, starting the surgery, and then rushing him back to stasis to finish the surgery. But you're right, this is kind of an issue depending on how much people want to fuck over plasmamen
> 
> Admittedly, said surgeries aren’t vital to MEDBAY, but they’re still nigh-impossible for plasmamen without quite a bit of setup beforehand.

**Comment by GuillaumePrata** [comment]
> Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.

You don't need a plasma room, just drag one of the public scrubbers to surgery and drain the O2 in the air.

**Comment by antropod** [comment]
pluoxium room

**Comment by jwc1015** [comment]
> > Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.
> 
> You don't need a plasma room, just drag one of the public scrubbers to surgery and drain the O2 in the air.

Doesn’t work; you need at LEAST the air alarm adjusted, otherwise more oxygen is added back in.

**Comment by Thalpy** [comment]
Okay, I speedcoded the reagent I mentioned before: https://github.com/tgstation/tgstation/pull/57315

In essence;
When applied to a dead (or living) mob it will slow down surgery steps by making them take double to triple the time (depending on the creation purity, of course!)
(To note, however, this reagent DOES process in the dead, and will be consumed over time, thus formaldehyde is better for longer term applications, or any patients that need a lot of surgery)

**Comment by LordVollkorn** [comment]
1.) patient dead? give an epipen or formaldehyde and use a table
2.) patient critical? give an epipen or epinephrine and use a table
3.) patient so critical he dies again within seconds? Go to 1, fix more or if you are overwhelmed use a stasis bed

isnt this how it was supposed to be?

**Comment by Wallemations** [comment]
While I understand that in their current state the stasis beds are a bit too powerful, I feel like a more interesting solution would be making it so instead of simply being able to link the operating computer at the start of a round, it would take an upgrade to the stasis beds' stock parts in order to be capable of upgrading. 

Stasis beds are one of many machines which require stock parts, but don't benefit from upgrades, and I feel like this is a fairly good way of having stasis beds stock parts be able to upgrade while still nerfing their ability to link to operating computers.

**Comment by unit0016** [comment]
_laughs in medical cyborg_

**Review by Mothblocks — APPROVED** [review]
alright, i'm gonna do the operating table thing tho

**Comment by bluetshirtguy** [comment]
I just liked the addition of computers working with statis beds, learning medical is very slow, stressful and having people harp about how terrible you are isn't fun either. For the sake of new players needing plenty of time please reconsider.

**Comment by Krysonism** [comment]
> 
> 
> Uh, nobody has mentioned this, but this makes plasmaman surgery near impossible without a plasma room set up; if ANY surgery besides basic stuff is needed then medbay will have to get atmos to make one, which is also easy to sabotage and I’m not sure if medbay would even bother most of the time.

That is not true, you can use a shower to keep the flames down.

**Comment by Mothblocks** [comment]
anyone saying plasmaman surgery is impossible is a zoomer who was not here before stasis beds existed.

stasis beds basically eliminate the entire challenge of reviving a plasmaman.

also, no part of reviving a plasmaman requires an advanced surgery.

**Comment by bluetshirtguy** [comment]
Very disappointing, I don't like the trend of making jobs harder simply because some players are already proficient. The strains you people continuously put on learners turns people away from learning medical.

**Comment by Mothblocks** [comment]
It's not that--it's that stasis beds have historically been really overpowered. There is effectively no reason to use a surgical table over a stasis bed. There's not many nerfs we can take to stasis beds, nerfing speed further isn't going to make most people use something else, and it's just a loss for the patient. Buffing everything around it is just furthering power creep. Really, this is one of the few decently impactful nerfs that could've been taken without neutering it.

**Comment by bluetshirtguy** [comment]
It was a godsend for new players in a very hard learning curve in a job where taking too long is enough for insults instead of thanks.

**Comment by ExcessiveUseOfCobblestone** [comment]
newbies arent going to know about operating computer linking anyways though?

Its not a crutch if it is better than the intended surgery bed, its just a straight roundstart upgrade. This nerf puts it in the crutch state.

**Comment by adamkad1** [comment]
> It's not that--it's that stasis beds have historically been really overpowered. There is effectively no reason to use a surgical table over a stasis bed. There's not many nerfs we can take to stasis beds, nerfing speed further isn't going to make most people use something else, and it's just a loss for the patient. Buffing everything around it is just furthering power creep. Really, this is one of the few decently impactful nerfs that could've been taken without neutering it.

back to being a borg then

**Comment by iprice** [comment]
> Medical literally always moves the computers to the stasis beds, bypassing operating tables almost completely. 

Medical mostly doesn't bother with the operating computers any more.  Failed to reach objective.

**Comment by Time-Green** [comment]
Or they just place both an operating console AND a surgical bed next to the stasis beds, which is meh

**Comment by iprice** [comment]
I think the bottom line is this.  You can't really make "useful/common" surgery require a fancy bed because you'll ruin the versatility of surgery and ghetto surgery.  So all this is really changing is advanced surgery, of which advanced tend wounds is all that we've lost here that mattered.

Otherwise, operating computers basically had a nice bar graph for frequent status updates on the patient, as well as step by step guidance, helpful mainly for newer players.  The nerf just removed two "qualities of life" and for the regulars this doesn't change stasis beds.

Actually killing off stasis beds makes the just suck by adding constant time demands against bleeding out, toxins, plasma people problems and so on.

I'm not really sure why stasis "needs to be nerfed" or "operating tables need to be used" (opinion as fact) and this just doesn't do it.
